### PR TITLE
PS-8101 Documentation for Thread Pool functionality in 5.7 has duplic…

### PR DESCRIPTION
**Pull request by patrickbirch**
…ated text (5.7)

 M source/performance/threadpool.rst

### CODE REVIEW

File: doc/source/performance/threadpool.rst [hunk 1]
```diff
@@ -4,19 +4,7 @@
...
-the context switching and hot locks contentions. Using the thread pool will have
-the most effect with ``OLTP`` workloads (relatively short CPU-bound queries).
-
-The |thread-pool| feature introduces a dynamic |thread-pool| that enables the
-server to keep the top performance even with a large number of client
-connections. With |thread-pool| enabled, the server decreases the number of
-threads, which then reduces the context switching and hot locks
-contentions. Using the |thread-pool| has the most effect on ``OLTP`` workloads
-(relatively short CPU-bound queries).
+*MySQL* executes statements using one thread per client connection.  Th performance degrades when the number of connections increase past a certain point. A dynamic thread pool enables the server to keep the top performance even with a large number of client connections. The thread pool server decreases the number of threads, which reduces the context switching and hot locks contentions. Thread pools make the ``OLTP`` workloads (relatively short CPU-bound queries) more efficient.
```
Comment (nastena1606) [inline doc/source/performance/threadpool.rst]:
```suggestion
*MySQL* executes statements using one thread per client connection.  The performance degrades when the number of connections increases past a certain point. A dynamic thread pool enables the server to keep the top performance even with a large number of client connections. The thread pool server decreases the number of threads, which reduces the context switching and hot locks contentions. Thread pools make the ``OLTP`` workloads (relatively short CPU-bound queries) more efficient.
```

Comment (patrickbirch) [reply doc/source/performance/threadpool.rst]: Fixed.

**Review by nastena1606 — APPROVED** [review]

**Review by alina-derkach-oaza — APPROVED** [review]

File: doc/source/performance/threadpool.rst [hunk 2]
```diff
@@ -4,19 +4,17 @@
  Thread Pool
 =============
 
+Servers continually execute queries from multiple clients. In *MySQL*, for each connection query, the server creates a thread, processes the query, and then destroys the thread. This method can have disadvantages because the server must consume resources along with the consuming the resources needed to process the query. Therefore, when the number of connections grows the server performance drops. Too many active threads impact performance because of context switching, and thread contention.
```
Comment (alina-derkach-oaza) [inline doc/source/performance/threadpool.rst]: "This method can have disadvantages because" -> I would remove this part

Comment (alina-derkach-oaza) [reply doc/source/performance/threadpool.rst]: along with the consuming the resources -> along with consuming the resources

Comment (alina-derkach-oaza) [reply doc/source/performance/threadpool.rst]: Therefore, when the number of connections grows the server performance drops. -> remove the comma after "Therefore" and add a comma after "grows"

File: doc/source/performance/threadpool.rst [hunk 3]
```diff
@@ -4,19 +4,17 @@
...
-(relatively short CPU-bound queries).
+A thread pool is distinct from connection pooling. A thread pool has the following advantages:
+
+* Limits the number of threads running on the server
+
+* Minimizes wasting resources by creating and then destroying threads
+
+This feature ensures that multiple connections using a thread pool will not cause the server to churn through resources or cause a server exit when the server runs out of memory. A thread pool reuses threads and is most efficient for the short queries associated with transactions. 
+
+If, for example, the server has 100 clients connecting and attempting to run queries, only a fraction of those queries are completed. If the server has a thread pool size of 10 and 100 clients connect to run queries, the thread pool efficiently queues and executes 10 queries at a time in parallel.
```
Comment (alina-derkach-oaza) [inline doc/source/performance/threadpool.rst]: a fraction of those queries IS completed

Comment (patrickbirch) [reply doc/source/performance/threadpool.rst]: Redone.

File: doc/source/performance/threadpool.rst (hunk 2 above)
Comment (patrickbirch) [reply doc/source/performance/threadpool.rst]: Keeping the disadvantages
fixed
fixed.

**Review by alina-derkach-oaza — APPROVED** [review]

**Pull request merged**